### ISSUE TITLE
Stack trace when ostree operates in filesystem not supporting extended attributes

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#268]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/268)
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Feb 23, 2018, 12:46

When initializing an artifact cache on a volume that does not support extended attributes, like a tmpfs, instead of a human readable error message, BuildStream aborts with a stack trace:

```
Traceback (most recent call last):
  File "/codethink/GNOME/buildstream/tests/testutils/runcli.py", line 287, in invoke
    cli.main(args=args or (), prog_name=cli.name, **extra)
  File "/codethink/GNOME/buildstream/buildstream/_frontend/cli.py", line 104, in override_main
    standalone_mode=standalone_mode, **extra)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/codethink/GNOME/buildstream/buildstream/_frontend/cli.py", line 203, in build
    fetch_subprojects=True)
  File "/codethink/GNOME/buildstream/buildstream/_frontend/main.py", line 188, in initialize
    rewritable=rewritable)
  File "/codethink/GNOME/buildstream/buildstream/_pipeline.py", line 120, in __init__
    Platform._create_instance(context, project)
  File "/codethink/GNOME/buildstream/buildstream/_platform/platform.py", line 63, in _create_instance
    cls._instance = PlatformImpl(*args, **kwargs)
  File "/codethink/GNOME/buildstream/buildstream/_platform/linux.py", line 39, in __init__
    self._artifact_cache = OSTreeCache(context, self._user_ns_available)
  File "/codethink/GNOME/buildstream/buildstream/_artifactcache/ostreecache.py", line 74, in __init__
    self.repo = _ostree.ensure(ostreedir, False)
  File "/codethink/GNOME/buildstream/buildstream/_ostree.py", line 54, in ensure
    repo.create(mode)
GLib.GError: g-io-error-quark: fsetxattr(user.ostreemeta): Operation not supported (15)
```

This should be caught and an `ArtifactError` with a reasonable error message should be raised instead.

This was discovered at the same time as #267

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 10, 2018, 08:30

mentioned in commit a615e00185b097a3ff3ce15f58e79a803339d9f8

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 10, 2018, 08:51

mentioned in commit 76cbcd4ec52c26fc77134533df64c64221610e22

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 10, 2018, 17:06

mentioned in commit 8c7ab2f0a3582cde19d5be7f293b47bfa51efbc4

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 10, 2018, 18:12

mentioned in commit 1792150b63971e2c4335f785cb9c7da1243cfcc4

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 16, 2018, 15:33

mentioned in commit cb79b711f0c653c10d78e44f3154eee4f9d123b9

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 16, 2018, 17:25

mentioned in commit fdbc1e33f84fd45f6ac4527846fc3cc8a14d009b

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 17, 2018, 06:58

mentioned in commit 1f8b4aa290a908a697f008a29ea143a9320dd639

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 17, 2018, 08:59

closed via merge request !337

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 17, 2018, 08:59

closed via commit 1f8b4aa290a908a697f008a29ea143a9320dd639
